src/util/properties.rs

```diff
@@ -34,40 +43,46 @@ const PROP_NUM_ROWS: &'static str = "tikv.num_rows";
 const PROP_NUM_PUTS: &'static str = "tikv.num_puts";
 const PROP_NUM_VERSIONS: &'static str = "tikv.num_versions";
 const PROP_MAX_ROW_VERSIONS: &'static str = "tikv.max_row_versions";
-const PROP_NUM_ERRORS: &'static str = "tikv.num_errors";
+const PROP_SIZE_INDEX: &'static str = "tikv.size_index";
 
-#[derive(Clone, Debug, Default)]
+const PROP_SIZE_INDEX_DISTANCE: u64 = 4 * 1024 * 1024;
+
+#[derive(Default)]
 pub struct UserProperties {
```

Comment:
I don't think we need a `UserProperties` to combine several properties collectors, just add them to ColumnFamilyOptions one by one.